web/explorer/src/components/RuleMatchesTable.vue

```diff
@@ -0,0 +1,289 @@
+    }
+});
+
+const treeData = ref([]);
+const filters = ref({});
+const filterMode = ref("lenient");
+const sourceDialogVisible = ref(false);
+const currentSource = ref("");
+const expandedKeys = ref({});
+
+const menu = ref();
+const selectedNode = ref({});
+
+const contextMenuItems = computed(() => [
+    {
+        label: "View source",
+        icon: "pi pi-eye",
+        command: () => {
+            showSource(selectedNode.value.data.source);
+        }
+    },
+    {
+        label: "View rule in capa-rules",
+        icon: "pi pi-external-link",
+        target: "_blank",
+        url: selectedNode.value.url
+    },
+    {
+        label: "Lookup rule in VirusTotal",
+        icon: "vt-icon",
+        target: "_blank",
+        url: selectedNode.value.vturl
+    }
+]);
+
+const onRightClick = (event, instance) => {
+    if (instance.node.data.source) {
+        selectedNode.value = instance.node;
+        // contrust capa-rules url
+        selectedNode.value.url = `https://github.com/mandiant/capa-rules/blob/master/${instance.node.data.namespace || "lib"}/${instance.node.data.name.toLowerCase().replace(/\s+/g, "-")}.yml`;
+        // construct VirusTotal deep link
+        const behaviourSignature = `behaviour_signature:"${instance.node.data.name}"`;
+        selectedNode.value.vturl = `https://www.virustotal.com/gui/search/${encodeURIComponent(behaviourSignature)}/files`;
+
+        menu.value.show(event);
+    }
+};
+
+/*
+ * Expand node on click
+ */
+const onNodeSelect = (node) => {
+    const nodeKey = node.key;
+    const nodeType = node.data.type;
+
+    // We only expand rule and match locations, if not return
+    if (nodeType !== "rule" && nodeType !== "match location") return;
+
+    // If the node is already expanded, collapse it
+    if (expandedKeys.value[nodeKey]) {
+        delete expandedKeys.value[nodeKey];
+        return;
+    }
+
+    if (nodeType === "rule") {
+        // For rule nodes, clear existing expanded keys and set the clicked rule as expanded
+        // and expand the first (child) match by default
+        expandedKeys.value = { [nodeKey]: true, [`${nodeKey}-0`]: true };
+    } else if (nodeType === "match location") {
+        // For match location nodes, we need to keep the parent expanded
+        // and toggle the clicked node while collapsing siblings
+        const [parentKey] = nodeKey.split("-");
+        expandedKeys.value = { [parentKey]: true, [`${nodeKey}`]: true };
+    } else {
+        return;
+    }
+};
+
+// All available columns
+const visibleColumns = ref([
+    { field: "address", header: "Address" },
+    { field: "namespace", header: "Namespace" },
+    { field: "tactic", header: "ATT&CK Tactic" },
+    { field: "mbc", header: "Malware Behaviour Catalogue" }
+]);
+
+// Filter out the treeData for showing/hiding lib rules
+const filteredTreeData = computed(() => {
+    if (props.showLibraryRules) {
+        return treeData.value; // Return all data when showLibraryRules is true
```

Comment:
when enabled these only show up at the bottom of the page, can they adhere to the currently chosen sorting order with an easy update here?

Reply:
@mr-tz The reason they are shown at the bottom is because the table is pre-sorted with the namespace column, and lib rules have no namespaces (`null`). If you remove that sorting from the namespace column, then enabling the lib rules will not show the matches at the bottom but in default occurence order

Reply:
ah, I see - perfect, this works as intended then